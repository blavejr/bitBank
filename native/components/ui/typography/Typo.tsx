import { TextProps, Text } from "react-native";
import { tv, VariantProps } from "tailwind-variants";

const BBText = tv({
    base: '',
    defaultVariants: {
        color: 'primary',
    },
    variants: {
      color: {
        primary: 'text-typography-primary',
        secondary: 'text-typography-secondary',
        tertiary: 'text-typography-tertiary',
        danger: 'text-red-500',
        success: 'text-green-500',
        warning: 'text-yellow-500',
        info: 'text-blue-500',
      },
      size: {
        sm: 'text-sm',
        md: 'text-md',
        lg: 'text-lg',
        xl: 'text-xl',
        '2xl': 'text-2xl',
        '3xl': 'text-3xl',
        '4xl': 'text-4xl',
      },
      weight: {
        normal: 'font-normal',
        bold: 'font-bold',
      },
    },
});
  
type TypographyProps = TextProps & VariantProps<typeof BBText>;
  
export function Typography({ className, ...props }: TypographyProps) {
    return (
      <Text className={BBText({ className, ...props })} {...props} />
    );
  }