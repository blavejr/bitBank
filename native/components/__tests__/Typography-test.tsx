import * as React from 'react';
import renderer from 'react-test-renderer';

import { Typography } from '../ui/typography/Typo';

it(`renders correctly`, () => {
  const tree = renderer.create(<Typography>Snapshot test!</Typography>).toJSON();

  expect(tree).toMatchSnapshot();
});
