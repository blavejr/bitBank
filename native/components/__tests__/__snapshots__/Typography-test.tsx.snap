// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  className="text-typography-primary"
>
  Snapshot test!
</Text>
`;
